### ISSUE TITLE
Some question about Algorithm 1 in your paper.

### ISSUE THREAD

**Issue by KeyKy** [XHPlus/DSQ#2]
1. When k is large enough, is asymptotic function phi plus Dequantize function equal to a quantize-dequantize function? If so, why do you need the stage 3 ("Keep consistent with standard quantization") in your algorithm 1?
```
a_q = sgn(a_sq)
w_q = sgn(w_sq)
```
2. I think no matter what k is, stage 3 make DSQ equal to quantize-dequantize function. At this time, DSQ is not differentiable and network can only be optimized by STE. If so, the performance gain dose not come from making quantization differentiable. Maybe, the performance gain comes from lower bound and upper bound optimization. Bound or Range optimization can be implemented by PACT.
3. DSQ can detect which operation is not suitable for quantization. If an operation is not suitable for quantization, which mean it has a big Alpha. A big Alpha DSQ is the same as identity function which make the operation can not be optimized by low-bit calculations. So, the performance gain maybe comes from partially quantization.

Thanks.

**Comment by XHPlus** [comment]
1. In Figure 3.b, we show that if apha is small enough (equal to k large enough), the DSQ without sgn will behave the same as uniform quantization. However, we found that it is hard to directly push alpha to such a small magnitude during training. So to keep the consistence, we still apply a sgn after DSQ.
2. Yes. So after adding sgn, the function of DSQ is to adjust the gradient, which can serve as a better substitute for STE
3. As aforementioned, because we have applied the sgn after DSQ, so the result comes from real quantization.
